### PR TITLE
LevelDB review

### PR DESCRIPTION
**Pull request by benbierens**
Review comments by dryajov.

### CODE REVIEW

**Review by dryajov — DISMISSED** [review]
Lets rebase my latest cleanup  (master) which should address most of the result related issues and merge.

**Comment by benbierens** [comment]
This PR now fixes both the batch-put and the iterator dispose.

**Comment by benbierens** [comment]
This PR now also:
- Lets LevelDB deal with queries that end in wildcards.
- Fixes TypedDS tests for LevelDB.

File: datastore/leveldb/leveldbds.nim [hunk 1]
```diff
@@ -104,10 +116,18 @@ method query*(
...
 
-  iter.next = next
-  iter.dispose = proc(): Future[?!void] {.async.} =
+  proc dispose(): Future[?!void] {.async.} =
+    try:
+      dbIter.dispose()
+      return success()
+    except LevelDbException as e:
+      return failure("LevelDbDatastore.query -> dispose exception: " & $e.msg)
+    except Exception as e:
```
Comment (dryajov) [inline datastore/leveldb/leveldbds.nim]: We shouldn't catch `Exception` and if we do it should be re-raised as a `Defect`.

Comment (benbierens) [reply datastore/leveldb/leveldbds.nim]: Updated nim-leveldb to correctly show what can be raised by the iterator callbacks.
Adjusted leveldbds.nim to match.

**Comment by benbierens** [comment]
This one must wait. It will fail until https://github.com/codex-storage/nim-leveldb/pull/2 is merged and a new version is published and pulled in here.

**Comment by dryajov** [comment]
> This one must wait. It will fail until [codex-storage/nim-leveldb#2](https://github.com/codex-storage/nim-leveldb/pull/2) is merged and a new version is published and pulled in here.

merged!

**Review by dryajov — DISMISSED** [review]

**Comment by benbierens** [comment]
This one should be ready to merge.

**Review by dryajov — APPROVED** [review]

**Pull request merged**